tools/eslint-rules/snake-case-filenames.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @fileoverview enforces snake_case for filenames.
+ * @author Shobhit Chittora <chittorashobhit@gmail.com>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const path = require('path');
+
+module.exports = function(context) {
+  const snakeCaseRegex = new RegExp('^[a-z_]+$');
+
+  return {
+    'Program': function(node) {
+      const filename = path.resolve(context.getFilename());
+      const name = path.basename(filename, path.extname(filename));
+
+      if (snakeCaseRegex.test(name)) {
+        context.report(node, "Filename '{{name}}' is not in snake_case.", {
```

Comment:
Refer https://developers.google.com/web/updates/2015/01/ES6-Template-Strings to learn about string interpolation.
Line 22 should be like 
> context.report(node, \`Filename ${name} is not in snake_case.\`, {

Reply:
@aryannarora Check the eslint `context.report()` doc  - https://eslint.org/docs/3.0.0/developer-guide/working-with-rules-deprecated#contextreport.

Will change if other concur.